pkg/cmd/cli/secrets/add_secret_to_obj.go

```diff
@@ -181,7 +175,41 @@ func (o AddSecretOptions) AddSecrets() error {
 	return nil
 }
 
-func (o AddSecretOptions) getSecrets() ([]*api.Secret, error) {
+// TODO: when Secrets in kapi.ServiceAccount get changed to MountSecrets and represented by LocalObjectReferences, this can be
+// refactored to reuse the addition code better
+// addSecretsToServiceAccount adds secrets to the service account, either as pull secrets, mount secrets, or both.
+func (o AddSecretOptions) addSecretsToServiceAccount(serviceaccount *kapi.ServiceAccount) error {
+	newSecrets, err := o.getSecrets()
+	if err != nil {
+		return err
+	}
+	if len(newSecrets) == 0 {
+		return errors.New("no secrets to add")
+	}
+	newSecretNames := getSecretNames(newSecrets)
+
+	if !o.ForPull && !o.ForMount {
+		return errors.New("this type of secret type is not handled for ServiceAccounts")
+	}
+	if o.ForMount {
+		currentSecrets := getMountSecretNames(serviceaccount)
+		secretsToAdd := getNamedSecrets(newSecrets, newSecretNames.Difference(currentSecrets))
```

Comment:
Translating from the `secretName` to the specific `secret` in `newSecrets` is nontrivial